modules/apps_processing.py

```diff
@@ -0,0 +1,209 @@
+import sys
+import os
+import yaml
+import shutil
+import configparser
+import tarfile
+from io import StringIO
+
+
+class AppFilesProcessor:
+    """Class for handling local app files and configurations."""
+
+    def __init__(self, yml_path):
+        self.yml_path = yml_path
+
+    def _read_yaml(self) -> dict:
+        """Read and return the contents of a YAML file."""
+        file_path = self.yml_path
+        with open(file_path, "r") as file:
+            return yaml.safe_load(file)
```

Comment:
Getting rid of `_read_yaml()` function that's called only once. Reducing complexity.
Reading file already at object creation, no need to wait.

NOTE: if you want to keep using passed parameters instead of env vars, please define it here, not in `deploy.py` for readability and modularity reasons

```suggestion
    def __init__(self):
        if not "DEPLOYMENT_CONFIG" in os.environ:
            raise Exception(f"Error - Environment variable DEPLOYMENT_CONFIG does not exist.")
        yml_path = os.getenv("DEPLOYMENT_CONFIG") 
        # Read the file
        try:
            with open(yml_path, "r") as file:
                self.data = yaml.safe_load(file)
        except FileNotFoundError:
            raise Exception(f"File not found: {yml_path}")
        except yaml.YAMLError as e:
            raise Exception(f"Error parsing YAML file: {e}")
```